tests/select-input-test.js

```diff
@@ -0,0 +1,335 @@
+import { describe, it } from 'mocha'
+import React from 'react'
+import * as assert from 'assert'
+import Enzyme from 'enzyme'
+import Adapter from 'enzyme-adapter-react-16'
+import sinon from 'sinon'
+
+Enzyme.configure({ adapter: new Adapter() })
+
+import SelectInput from '../src/inputs/select-input.jsx'
+
+const exampleFormData = {
+  'select-typeahead': { value: '' },
+  'select-button': { value: '' }
+}
+const options = [
+  {
+    label: 'A',
+    value: 'a'
+  },
+  {
+    label: 'B',
+    value: 'b'
+  },
+  {
+    label: 'C',
+    value: 'c'
+  }
+]
+
+let value = 'c'
+
+const updateValue = (formId, fieldName, fieldValue) => {
+  value = fieldValue
+}
+
+describe('<SelectInput />', function() {
+  const exampleLabel = 'Example Select Input'
+  const typeaheadInput = (
+    <SelectInput
+      formData={ exampleFormData }
+      formId='example-form'
+      label={ exampleLabel }
+      name='select-typeahead'
+      onChange={ updateValue }
+      options={ options }
+      value={ value }
+    />
+  )
+  const buttonInput = (
+    <SelectInput
+      formData={ exampleFormData }
+      formId='example-form'
+      label={ exampleLabel }
+      name='select-button'
+      onChange={ updateValue }
+      options={ options.map(opt => opt.value) }
+      typeAhead={ false }
+      value={ value }
+    />
+  )
+
+  // make sure there are separate divs to attach each to
+  const root1 = document.createElement('div')
+  root1.id = 'root-1'
+  const root2 = document.createElement('div')
+  root2.id = 'root-2'
+  document.getElementById('root').appendChild(root1)
+  document.getElementById('root').appendChild(root2)
+
+  const wrapper = Enzyme.mount(typeaheadInput, { attachTo: document.getElementById('root-1') })
+  const wrapper2 = Enzyme.mount(buttonInput, { attachTo: document.getElementById('root-2') })
+
+  function resetWrappers() {
+    wrapper.unmount()
+    wrapper.mount(typeaheadInput, { attachTo: document.getElementById('root-1') })
+
+    wrapper2.unmount()
+    wrapper2.mount(buttonInput, { attachTo: document.getElementById('root-2') })
+  }
+
+  describe('tests set up with root for mounting', function() {
+    it ('is ready to test', function() {
+      assert.ok(document.getElementById('root'))
+      assert.ok(document.getElementById('select-typeahead'))
+      assert.ok(document.getElementById('select-button'))
+    })
+  })
+
+  describe('rendering correctly:', function() {
+    it ('renders without breaking', function() {
```

Comment:
I'm not sure these assertions are actually doing what you may have intended.  They should return an instance of a spy if the `componentDidMount` exists, which would mean the assertion will be true if `componentDidMount` exists on the instance.  Your also setting the componentDidMount as a spy, I assume that was intended?

Reply:
You're right. I apparently didn't finish this test. Thanks!